client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,61 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final boolean subscriberForRetention;
+    private final boolean autoPublishAtLastCheckpoint;
+
+    public enum ReaderGroupRetentionConfig {
+        /**
+         * There is no retention done by this {@link ReaderGroup}.
+         */
+        NO_CONSUMPTION_BASED_TRUNCATION,
+
+        /**
+         * This {@link ReaderGroup} can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream according to its read positions. These read positions have to be provided
+         * manually in the form of a {@link StreamCut}, see this method:
+         *
+         * {@link ReaderGroup#updateRetentionStreamCut(Stream, StreamCut)}
+         */
+        TRUNCATE_AT_USER_STREAMCUT,
+
+        /**
+         * This {@link ReaderGroup} can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream using the read positions from the {@link StreamCut}s generated during checkpointing.
```

Comment:
using read positions from the lastCheckpoint.

Reply:
Updated the javadoc.